src/main-view/content/shortcut-button.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 - 2021 José Expósito <jose.exposito89@gmail.com>
+ *
+ * This file is part of Touché.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation,  either version 3 of the License,  or (at your option)  any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful,  but  WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the  GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+const { GObject, Gtk, Gdk } = imports.gi;
+const MODIFIERS = ['Control_L', 'Alt_L', 'ISO_Level3_Shift', 'Control_R', 'Shift_L', 'Shift_R'];
+
+class ShortcutButton extends Gtk.Button {
```

Comment:
This is looking much better, thanks. However, there is still some areas where the `SendKeysRowSettings` needs to know too much about how `ShortcutButton` works internally.

`SendKeysRowSettings` shouldn't have `modifiersEntry` and `keysEntry`, as those widgets are not displayed.

Instead, the `ShortcutButton` widget could have a couple of methods `getModifiers` and `getKeys` that `SendKeysRowSettings` could check on `getSettings()`.

No signals (`modifier_addition`, `key_addition` and `require_clear_shortcut`) would be required with this approach and you could even remove `this.keys` and `this.modifiers` from `SendKeysRowSettings`.

In that way, the widget could be truly reusable and replacing `oppositeKeysEntry` should require minimal changes.

Let me know if I didn't explain myself and I'll show some code :smile:

Reply:
No problem, I understand. I will check it later today ;)

Reply:
I have just pushed new changes. All it's good now, I hope i don't have more developments to do now :D

Can you check and resolve/update all comments ? Hope you can merge it now